docs/internals/controllers.md

```diff
@@ -0,0 +1,31 @@
+# Controllers
+
+The main controller implementations ([cluster] and [machine]) are located
+within  the `cluster-api` library, and each perform almost identical basic
+steps within their `Reconcile` methods. The controllers have the responsibility
+of receiving incoming requests and dispatching them to the appropriate actuator
```

Comment:
It helps me to think of this in different terms than "a request is received from Kubernetes." Namel,y the controller _watches_ for changes to the resources. For example, the machine controller watches for changes to the machine resource. If a Machine object is created, updated, or deleted, the controller receives that object.

Reply:
Fair. I've fixed some language in this area to mention the watches, and fixed up the basic flow to mention that after a watch is observed, a reconcile request is received before the controller fetches the cluster/machine object from Kubernetes.

Reply:
Thanks a lot!